modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java

```diff
@@ -1357,8 +1357,12 @@ public MediaType getMediaType(ObjectNode contentNode, String location, ParseResu
 
 		ObjectNode encodingObject = getObject("encoding", contentNode, false, location, result);
 		if (encodingObject != null) {
-			mediaType.setEncoding(getEncodingMap(encodingObject, String.format("%s.%s", location, "encoding"),
-					result));
+            String encodingLocation = String.format("%s.%s", location, "encoding");
+			mediaType.setEncoding(getEncodingMap(encodingObject, encodingLocation, result));
+
+            mediaType.getEncoding().keySet().stream()
+                .filter( ep -> Optional.ofNullable( mediaType.getSchema().getProperties()).map( p -> !p.containsKey( ep)).orElse( true))
```

Comment:
This should be only executed for inline schemas, by skipping it when a schema `$ref` field is existing and non empty, as properties in this case can be defined in the referenced schema